zerver/views/user_groups.py

```diff
@@ -115,6 +119,54 @@ def update_user_group_backend(
     return json_success(request, data)
 
 
+def send_messages_for_user_groups(
+    acting_user: UserProfile,
+    recipient_users: List[UserProfile],
+    user_group_name: str,
+    *,
+    send_subscription_message: bool,
+    send_unsubscription_message: bool,
+) -> None:
+    realm = acting_user.realm
+    mention_backend = MentionBackend(realm.id)
+
+    notifications = []
+    sender = get_system_bot(settings.NOTIFICATION_BOT, realm.id)
+    for recipient_user in recipient_users:
+        if recipient_user.email == acting_user.email:
+            # Don't send notification message if you subscribed/unsubscribed yourself.
+            continue
+        if recipient_user.is_bot:
+            # Don't send notification message to bots.
+            continue
+
+        with override_language(recipient_user.default_language):
+            if send_subscription_message:
+                message = _("{user_full_name} subscribed you to the group {group_name}.").format(
+                    user_full_name=silent_mention_syntax_for_user(acting_user),
+                    group_name=f"@_*{user_group_name}*",
+                )
+            if send_unsubscription_message:
+                message = _(
+                    "{user_full_name} unsubscribed you from the group {group_name}."
```

Comment:
These should be "added you to / removed you from" -- we're reserving the word "subscribed" for stream membership.

(I know the original issue called for these strings, but I've checked with Alya and that was an oversight).